toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -315,7 +315,7 @@ func addOrUpdateUser(user imagecustomizerapi.User, baseConfigPath string, imageC
 		}
 
 		// Add the user.
-		err = userutils.AddUser(user.Name, hashedPassword, uidStr, imageChroot)
+		err = userutils.AddUser(user.Name, user.HomeDirectory, user.PrimaryGroup, hashedPassword, uidStr, imageChroot.(*safechroot.Chroot))
```

Comment:
You need to handle the home directory for the case where the user is updated instead of added.

If it isn't possible (or practical) to move the home directory, then you should report an error if the home directory value is specified.